src/spdx/parser/tagvalue/parser.py

```diff
@@ -0,0 +1,546 @@
+            self.current_element["logger"].append(
+                f"Error while parsing {p[1]}: Value did not match expected format. Line: {p.lineno(1)}")
+            return
+        value = match.group(verif_code_code_grp)
+        excluded_files = None
+        if match.group(verif_code_exc_files_grp):
+            excluded_files = match.group(verif_code_exc_files_grp).split(",")
+        self.current_element["verification_code"] = PackageVerificationCode(value, excluded_files)
+
+    @grammar_rule("files_analyzed : PKG_FILES_ANALYZED LINE")
+    def p_pkg_files_analyzed(self, p):
+        if not self.check_that_current_element_matches_class_for_value(Package, p.lineno(1)):
+            return
+        if "files_analyzed" in self.current_element:
+            self.current_element["logger"].append(f"Multiple values for {p[1]} found. Line: {p.lineno(1)}")
+            return
+        self.current_element["files_analyzed"] = p[2] in ['true', 'True']
+
+    @grammar_rule("primary_package_purpose : PRIMARY_PACKAGE_PURPOSE LINE")
+    def p_primary_package_purpose(self, p):
+        if self.check_that_current_element_matches_class_for_value(Package, p.lineno(1)):
+            set_value(p, self.current_element, method_to_apply=lambda x: PackagePurpose[x.replace("-", "_")])
+
+    @grammar_rule("built_date : BUILT_DATE DATE\n release_date : RELEASE_DATE DATE\n "
+                  "valid_until_date : VALID_UNTIL_DATE DATE")
+    def p_package_dates(self, p):
+        if self.check_that_current_element_matches_class_for_value(Package, p.lineno(1)):
+            set_value(p, self.current_element, method_to_apply=datetime_from_str)
+
+    # parsing methods for snippet
+
+    @grammar_rule("snippet_attribution_text : SNIPPET_ATTRIBUTION_TEXT text_or_line")
+    def p_snippet_attribution_text(self, p):
+        if self.check_that_current_element_matches_class_for_value(Snippet, p.lineno(1)):
+            self.current_element.setdefault("attribution_texts", []).append(p[2])
+
+    @grammar_rule("snippet_license_info : SNIPPET_LICENSE_INFO license_or_no_assertion_or_none")
+    def p_snippet_license_info(self, p):
+        if not self.check_that_current_element_matches_class_for_value(Snippet, p.lineno(1)):
+            return
+        if p[2] == SpdxNone() or p[2] == SpdxNoAssertion():
+            self.current_element["license_info_in_snippet"] = p[2]
+        else:
+            self.current_element.setdefault("license_info_in_snippet", []).append(p[2])
+
+    @grammar_rule("snippet_byte_range : SNIPPET_BYTE_RANGE LINE\n snippet_line_range : SNIPPET_LINE_RANGE LINE")
+    def p_snippet_range(self, p):
+        if not self.check_that_current_element_matches_class_for_value(Snippet, p.lineno(1)):
+            return
+
+        argument_name = TAG_DATA_MODEL_FIELD[p[1]][1]
+        if argument_name in self.current_element:
+            self.current_element["logger"].append(
+                f"Multiple values for {p[1]} found. Line: {p.lineno(1)}")
```

Comment:
This is an error, so we should return here, don't we? Otherwise we will just overwrite the current value.

Reply:
correct